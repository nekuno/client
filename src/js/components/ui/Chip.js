import PropTypes from 'prop-types';
import React, { Component } from 'react';

export default class Chip extends Component {

    static propTypes = {
        label         : PropTypes.string,
        disabled      : PropTypes.bool,
        value         : PropTypes.string,
        onClickHandler: PropTypes.func,
        chipClass     : PropTypes.string
    };

    constructor(props) {

        super(props);
        this.handleClick = this.handleClick.bind(this);
    }

    handleClick(event) {
        if (typeof this.props.onClickHandler == 'function') {
            this.props.onClickHandler(this.props.value, event);
        }
    }

    render() {
        const {disabled, label} = this.props;
        let chipClass = this.props.chipClass ? this.props.chipClass + ' chip' : 'chip';
        return (
            <div className={disabled ? chipClass + " disabled-chip" : chipClass} onClick={this.handleClick}>
                <div className="chip-label">
                    {label ? label : this.props.children}
                </div>
            </div>
        );
    }
}
