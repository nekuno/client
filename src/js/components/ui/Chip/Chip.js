import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './Chip.scss';

export default class Chip extends Component {

    static propTypes = {
        onClickHandler : PropTypes.func.isRequired,
        onCancelHandler: PropTypes.func,
        text           : PropTypes.string.isRequired,
        selected       : PropTypes.bool
    };

    handleClick() {
        this.props.onClickHandler();
    }

    handleCancel() {
        if (this.props.onCancelHandler) {
            this.props.onCancelHandler();
        }
    }

    render() {
        const {text, selected} = this.props;
        const chipClass = selected ? styles.selected + ' ' + styles.chip : styles.chip;

        return (
            <div className={chipClass}>
                {this.props.onCancelHandler ?
                    <span className={styles.cancelIcon + ' icon icon-cancel-circle'} onClick={this.handleCancel.bind(this)}/>
                    : null
                }
                <div className={styles.text} onClick={this.handleClick.bind(this)}>{text}</div>
            </div>
        );
    }
}