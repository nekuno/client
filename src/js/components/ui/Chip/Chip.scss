.chip {
  background: #FFF;
  font-weight: 400;
  display: inline-block;
  height: 32px;
  line-height: 32px;
  border-radius: 16px;
  -moz-border-radius: 16px;
  -webkit-border-radius: 16px;
  padding: 0 12px 0 7px;
  box-sizing: border-box;
  vertical-align: middle;
  display: -webkit-inline-box;
  display: -ms-inline-flexbox;
  display: -webkit-inline-flex;
  display: inline-flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-align-items: center;
  align-items: center;
  margin: 2px 0;
  max-width: 400px;
  font-size: 12px;

  &.selected {
    background: #DDDDDD;
  }
  .cancel-icon {
    color: #999999;
    cursor: pointer;
    font-size: 20px;
  }
  .text {
    cursor: pointer;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    position: relative;
    -webkit-box-flex: 1;
    -webkit-flex-shrink: 1;
    -ms-flex: 0 1 auto;
    flex-shrink: 1;
    min-width: 0;
    margin-left: 6px;
  }
}