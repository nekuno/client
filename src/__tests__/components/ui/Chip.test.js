import React from 'react';
import Chip from '../../../js/components/ui/Chip/Chip.js';
import styles from '../../../js/components/ui/Chip/Chip.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test Chip component', () => {
    const mockFn = jest.fn();
    const mockFnCancel = jest.fn();
    const chipUnselected = shallow((<Chip onClickHandler={mockFn} onCancelHandler={mockFnCancel} text={'Lorem ipsum dolor sit amet'} selected={false} />));
    const chipSelected = shallow((<Chip onClickHandler={mockFn} onCancelHandler={mockFnCancel} text={'Lorem ipsum dolor sit amet'} selected={true} />));

    it('should be defined', () => {
        expect(Chip).toBeDefined();
    });
    it('should render correctly', () => {
        expect(chipUnselected).toMatchSnapshot();
        expect(chipSelected).toMatchSnapshot();
    });
    it('Chip unselected click event', () => {
        chipUnselected.find(`.${styles.text}`).simulate('click');
        expect(mockFn).toHaveBeenCalledTimes(1);
    });
    it('Chip selected click event', () => {
        chipSelected.find(`.${styles.text}`).simulate('click');
        expect(mockFn).toHaveBeenCalledTimes(1);
    });
    it('Chip unselected cancel event', () => {
        chipUnselected.find(`.${styles.cancelIcon}`).simulate('click');
        expect(mockFnCancel).toHaveBeenCalledTimes(1);
    });
    it('Chip selected cancel event', () => {
        chipSelected.find(`.${styles.cancelIcon}`).simulate('click');
        expect(mockFnCancel).toHaveBeenCalledTimes(1);
    });
});
