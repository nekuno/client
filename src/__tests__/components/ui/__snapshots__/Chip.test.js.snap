// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Chip component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Chip
    onCancelHandler={[MockFunction]}
    onClickHandler={[MockFunction]}
    selected={false}
    text="Lorem ipsum dolor sit amet"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="cancelIcon icon icon-cancel-circle"
          onClick={[Function]}
        />,
        <div
          className="text"
          onClick={[Function]}
        >
          Lorem ipsum dolor sit amet
        </div>,
      ],
      "className": "chip",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "cancelIcon icon icon-cancel-circle",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem ipsum dolor sit amet",
          "className": "text",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Lorem ipsum dolor sit amet",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="cancelIcon icon icon-cancel-circle"
            onClick={[Function]}
          />,
          <div
            className="text"
            onClick={[Function]}
          >
            Lorem ipsum dolor sit amet
          </div>,
        ],
        "className": "chip",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "cancelIcon icon icon-cancel-circle",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem ipsum dolor sit amet",
            "className": "text",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Lorem ipsum dolor sit amet",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Test Chip component should render correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Chip
    onCancelHandler={[MockFunction]}
    onClickHandler={[MockFunction]}
    selected={true}
    text="Lorem ipsum dolor sit amet"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="cancelIcon icon icon-cancel-circle"
          onClick={[Function]}
        />,
        <div
          className="text"
          onClick={[Function]}
        >
          Lorem ipsum dolor sit amet
        </div>,
      ],
      "className": "selected chip",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "cancelIcon icon icon-cancel-circle",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem ipsum dolor sit amet",
          "className": "text",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Lorem ipsum dolor sit amet",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="cancelIcon icon icon-cancel-circle"
            onClick={[Function]}
          />,
          <div
            className="text"
            onClick={[Function]}
          >
            Lorem ipsum dolor sit amet
          </div>,
        ],
        "className": "selected chip",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "cancelIcon icon icon-cancel-circle",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem ipsum dolor sit amet",
            "className": "text",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Lorem ipsum dolor sit amet",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
